src/drivers/intel/cavs/ssp.c

```diff
@@ -38,8 +38,22 @@ DECLARE_SOF_UUID("ssp-dai", ssp_uuid, 0x31458125, 0x95c4, 0x4085,
 /* empty SSP transmit FIFO */
 static void ssp_empty_tx_fifo(struct dai *dai)
 {
+	int ret;
 	uint32_t sssr;
 
+	/*
+	 * wait for set TNF then for TFL zero - order matter.
+	 * For 8000 Hz rate one sample (at least 2 bytes)
+	 * is transmitted within 125us
+	 */
+	ret = poll_for_register_delay(dai_base(dai) + SSSR, SSSR_TNF, SSSR_TNF,
+				      200);
+	ret |= poll_for_register_delay(dai_base(dai) + SSCR3, SSCR3_TFL_MASK, 0,
+				       125 * (SSP_FIFO_DEPTH - 1) / 2);
```

Comment:
can we avoid the magic numbers here? Please use defines.

Also I am a bit puzzled by the two successive delays, do we really need to wait for that long?

Reply:
Changed to defines, SSSR_TNF is cleared when TX FIFO is empty or full, it's why there are two polling functions